dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/oracle/AbstractOracleConnector.java

```diff
@@ -190,15 +176,46 @@ Properties buildProperties(@Nonnull ConnectorArguments arguments) {
   // jdbc:oracle:thin:<TNSName> (from 10.2.0.1.0)
   static String buildUrl(ConnectorArguments arguments) {
     String url = arguments.getUri();
-    if (url == null) {
-      String host = arguments.getHost();
-      int port = arguments.getPort(OPT_PORT_DEFAULT);
-      if (isOracleSid(arguments)) {
-        url = "jdbc:oracle:thin:@" + host + ":" + port + ":" + arguments.getOracleSID();
-      } else {
-        url = "jdbc:oracle:thin:@//" + host + ":" + port + "/" + arguments.getOracleServicename();
-      }
+    String host = arguments.getHost();
+    int port = arguments.getPort(OPT_PORT_DEFAULT);
+    if (url != null) {
+      checkNonUriFlags(arguments);
+      return url;
+    }
+    checkServiceAndSid(arguments);
+    if (arguments.getOracleSID() != null) {
+      return "jdbc:oracle:thin:@" + host + ":" + port + ":" + arguments.getOracleSID();
+    } else {
+      return "jdbc:oracle:thin:@//" + host + ":" + port + "/" + arguments.getOracleServicename();
+    }
+  }
+
+  private static void checkNonUriFlags(ConnectorArguments arguments) {
+    if (arguments.getHost() != null) {
+      throw extraFlagProvided("host");
+    } else if (arguments.getPort() != null) {
+      throw extraFlagProvided("port");
+    } else if (arguments.getOracleServicename() != null) {
+      throw extraFlagProvided("oracle-service");
+    } else if (arguments.getOracleSID() != null) {
+      throw extraFlagProvided("oracle-sid");
     }
-    return url;
+  }
+
+  private static void checkServiceAndSid(ConnectorArguments arguments) {
+    boolean hasService = arguments.getOracleServicename() != null;
+    boolean hasSid = arguments.getOracleSID() != null;
+    if (!hasService && !hasSid || hasService && hasSid) {
+      throw new MetadataDumperUsageException(
+          "Provide either -oracle-service or -oracle-sid for oracle dumper");
+    }
+  }
+
+  private static MetadataDumperUsageException extraFlagProvided(String flagName) {
+    String message =
+        String.format(
+            "Both url and %s were provided. If the url is valid, please omit the %s",
```

Comment:
I would make the message more balanced, so that it is not biased towards one or the other. Instead of "If the url is valid, please omit...", I would suggest something along the lines of "Use either X, or Y.".

Reply:
Done